docs/guides/using-the-network-shim.md

```diff
@@ -0,0 +1,288 @@
+
+### Cypress
+
+#### Environment variables
+
+The Network Shim depends on a few cypress environment variables. These can be set using the `--env` flag when interacting with the cypress CLI, but base values should be provided as well in via the following configuration files, see the [cypress docs](https://docs.cypress.io/guides/guides/environment-variables) for more information.
+
+##### `./cypress.json`
+
+This file is meant to be checked into source control, so should not contain any sensitive or machine-specific information.
+
+```json
+{
+    "baseUrl": "APP_URL",
+    "projectId": "PROJECT_ID",
+    "testFiles": "TEST_FILES",
+    "experimentalInteractiveRunEvents": true,
+    "video": false,
+    "env": {
+        "dhis2DataTestPrefix": "DATA_TEST_PREFIX",
+        "networkMode": "live", // default mode for the Network Shim
+        "dhis2ApiVersion": "value_from_prompt"
+    }
+}
+```
+
+_Notes:_
+
+-   _Only the `env` fields are required directly by the Network Shim, but other fields are included in this example since they are required to have a working cypress setup._
+-   _Interactive runs are not formally supported by the Network Shim, because `cypress open` (interactive mode) and `cypress run` have different behaviour regarding the plugin events. However, cypress is attempting to smooth things over and by setting `experimentalInteractiveRunEvents` to `true` it should theoretically be possible to at at least use stub mode in interactive mode._
+
+##### `./cypress.env.json`
+
+This file is meant to be git-ignored and we can use this to set environment variable that may contain sensitive or machine-dependent information.
+
+```json
+{
+    "dhis2Username": "USERNAME", // secret
+    "dhis2Password": "PASSWORD", // secret
+    "dhis2BaseUrl": "DHIS2_BASE_URL" // machine specific
+}
+```
+
+_Note that `dhis2Username` and `dhis2Password` are not required directly by the Network Shim, but by the `enableAutologin` setup. They are included in this example since they will be needed for most use-cases._
+
+#### Plugins file
+
+The `networShim` plugin should be imported and initialised in the cypress plugins file, see the [cypress docs](https://docs.cypress.io/guides/tooling/plugins-guide#Using-a-plugin) for more info. The plugin function doesn't require the second `config` argument most plugins accept.
+
+```js
+const { networkShim } = require('@dhis2/cypress-plugins')
+
+module.exports = (on, config) => {
+    networkShim(on)
+}
+```
+
+The `networkShim` plugin also accepts a second `options` argument. For most DHIS2 platform apps the default options will be correct so there will be no need to pass a second argument.
+
+Also for most DHIS2 cypress setups, using the `enableAutoLogin` setup helps managing authentication conveniently. In turn, the `enableAutoLogin` setup utility requires the `chromeAllowXSiteCookies` when using Chrome. So in most cases it makes sense to always use this plugin alongside the `networkShim`.
```

Comment:
Tweaked it a little:

> For most DHIS2 Cypress test suites the `enableAutoLogin` plugin can help reduce repetitive login steps in tests. The `enableAutoLogin` setup utility requires the `chromeAllowXSiteCookies` flag to be set when using Chrome for testing.

This sentence:

> So in most cases it makes sense to always use this plugin alongside the `networkShim`.

Doesn't follow clearly for me from the sentences before it. Why does that make sense (to always use this plugin)?

Reply:
Used a bit of mine and a bit of your suggestion here in the end.